core/src/main/java/com/linecorp/armeria/client/RedirectingClient.java

```diff
@@ -274,17 +290,127 @@ private void execute0(ClientRequestContext ctx, RedirectContext redirectCtx,
         });
     }
 
+    @Nullable
+    @VisibleForTesting
+    static RequestTarget resolveLocation(ClientRequestContext ctx, String location) {
+        final long length = location.length();
+        assert length > 0 : location;
```

Comment:
```suggestion
        assert length > 0;
```
It seems like the `location` doesn't provide additional information when it's empty.